packages/yarnpkg-core/sources/Manifest.ts

```diff
@@ -363,7 +364,7 @@ export class Manifest {
           continue;
         }
 
-        if (typeof range !== `string` || !semverUtils.validRange(range)) {
+        if (typeof range !== `string` || (!semverUtils.validRange(range) && !range.startsWith(WorkspaceResolver.protocol))) {
```

Comment:
```suggestion
        if (typeof range !== `string` || (!range.startsWith(WorkspaceResolver.protocol) && !semverUtils.validRange(range))) {
```
Should be faster to check the protocol first, especially with a lot of workspaces